prime-router/docs/proposals/0011-internal-formats/0011-internal-formats.md

```diff
@@ -0,0 +1,101 @@
+
+### Initial Effort
+The initial effort will be high for all options, essentially causing rewrites to all serializers, transforms, and filters. However, ReportStream will have more HL7 use-cases in the short term, so either a custom class or HL7 approach will be the easiest to get started. 
+
+### Summary
+The table below grades each option on a 1-5 scale, where five is best. 
+
+|            | New Use-Cases | New People | Guardrails  | Initial Effort  |
+|:----       |:----------:   |:----------:|:-----------:|:---------------:|
+| HL7        | 3             | 3          | 2           | 4               |
+| FHIR       | 5             | 4          | 5           | 2               |
+| Custom     | 4             | 2          | 1           | 5               |
+
+## Recommendation
+Although an HL7 internal format would be the quickest path to support full ELR, since ELR is an HL7-based standard, this path would not support the future FHIR scenarios. HL7 also has problems representing new data elements like those that SimpleReport uses to capture new testing sites. Although it may take more initial work, ReportStream should use FHIR as its internal structure because it supports more future scenarios and benefits engineers.  
+
+## Details
+
+### Topics
+In this proposal, the concept of a topic becomes something like parts of a FHIR implementation guide which defines what FHIR resources are in a particular bundle. Further investigation should be done to see if they are the same concept. 
+
+### Data elements
+
+Similar to the XPath concept for XML, FHIR defines a path syntax for its data elements. The FHIR path replaces the data element name in most cases in the current engine. The following table shows how some of the current data elements will translate into FHIR paths. 
+
+| Current Name          | FHIR Path   | Comments | 
+|:---------|:----------|:-----|
+| patient_last_name     | Patient.name.family   | FHIR follows USCDI naming conventions  | 
+| patient_first_name    | Patient.name.given[0] | FHIR supports multiple given names |
+| specimen_source_site  | Specimen.collection.bodySite | SNOMED-CT value |
+| ordering_provider_npi | ServiceRequest.requester.identifier | NPI value with CMS as the assigner |
+| site_of_care | ServiceRequest.location.extension.siteOfCare | Site of care is an extension on the location of the request |
+
+### Proposal 10 - Translations and Schemas
+This proposal can be thought of as an extension to proposal 10, which reduces the role of a schema. This proposal says the internal format is the FHIR standard implementation guide for a particular topic, thus eliminating the schema altogether. 
+
+### Mappers and transforms
+
+In combination with proposal #10, the Mappers types can be replaced by whole message transforms. 
```

Comment:
I think @whytheplatypus is working on a proposal that elucidates a path forwards on this, close to what you've outlined in this and the other Proposal.